### PR TITLE
[Fleet] added missing privileges to fleet-server-remote account

### PR DESCRIPTION
**Pull request by juliaElastic**
Related to https://github.com/elastic/kibana/issues/173237

Since we are opening up remote elasticsearch output feature in Fleet for integration data, agents have to get access to `traces-*` .

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? yes
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)? yes
- If submitting code, have you built your formula locally prior to submission with `gradle check`? yes
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)? yes
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by juliaElastic** [comment]
Hi @elastic/es-core-infra team, can I get your eyes on this please?

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ final class ElasticServiceAccounts {
             new String[] { "monitor", "manage_own_api_key" },
             new RoleDescriptor.IndicesPrivileges[] {
                 RoleDescriptor.IndicesPrivileges.builder()
-                    .indices("logs-*", "metrics-*")
+                    .indices("logs-*", "metrics-*", "traces-*")
```
Comment (jsoriano) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java:160]: From the description it looks like we would also need to add something for `.logs-endpoint*`?

**Review by jsoriano — APPROVED** [review]

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java (hunk 1 above)
Comment (jsoriano) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java:160]: Oh, I see it was added but reverted as not needed https://github.com/elastic/elasticsearch/pull/103445/commits/c88a773891106080decbafef01550a34d00a1afd

Comment (juliaElastic) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java:160]: yes, updated the description

**Pull request merged**